src/ToDo.js

```diff
@@ -1,21 +1,29 @@
 import React from "react";
 
 const ToDo = ({ todo, handleToggle }) => {
+  const [checked, setChecked] = React.useState(todo.complete);
   const handleClick = (e) => {
+    setChecked(e.currentTarget.checked);
     e.preventDefault();
     handleToggle(e.currentTarget.id);
   };
 
+  const handleChange = (event) => {
+    setChecked(event.target.checked);
+  };
+
   return (
     <div
-      id={todo.id}
+      value={todo.id}
       key={todo.id + todo.task}
+      id={todo.id}
       name="todo"
-      value={todo.id}
       onClick={handleClick}
-      className={todo.complete ? "todo strike" : "todo"}
     >
-      {todo.task}
+      <input type="checkbox" checked={checked} onChange={handleChange} />
```

Comment:
the `handleChange` function is not necessary. As per the current design, the todo state is handled at the `App.js` layer 

https://github.com/SanjanaPurma/sample-react-proj/blob/b99526d9082d8473fd21f418c18fe46669377ee5/src/App.js#L12-L19

Hence, when checkbox click, you can trigger the same function to update the flag.